sdk/metric/config.go

```diff
@@ -155,6 +160,21 @@ func WithExemplarFilter(filter exemplar.Filter) Option {
 	})
 }
 
+// WithCardinalityLimit sets the cardinality limit for the MeterProvider.
+//
+// The cardinality limit is the hard limit on the number of metric streams
+// that can be collected for a single instrument.
+//
+// By default, the cardinality limit is set to 2000.
+//
+// Setting this to a zero or negative value means no limit is applied.
+func WithCardinalityLimit(limit int) Option {
```

Comment:
can we leave this unexported until the feature is implemented?

Reply:
Leaving this unexported means we and more work resolving #6887, right? That issue is tracked in the milestone for the next release. 

If this is not implemented I'm the release and we move the issue to the next can't we just revert this then?

Reply:
Thats also fine by me